backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```diff
@@ -1211,6 +1216,146 @@ public MessageHeader createMessageHeader(
+
+    diagnosticCoding.setSystem(LOINC_CODE_SYSTEM).setCode(props.getTestPerformedCode());
+    diagnosticReport.setCode(diagnosticCodeableConcept);
+
+    var effectiveDateTime = DateTimeUtils.convertToZonedDateTime(props.getTestEffectiveDate());
+    diagnosticReport.setEffective(convertToDateTimeType(effectiveDateTime));
+
+    String diagnosticReportId = uuidGenerator.randomUUID().toString();
+    diagnosticReport.setId(diagnosticReportId);
+
+    return diagnosticReport;
+  }
+
+  public Observation convertToObservation(ConditionAgnosticConvertToObservationProps props) {
+    Observation observation = new Observation();
+    setStatus(observation, props.getCorrectionStatus());
+
+    var labCodeableConcept = new CodeableConcept();
+    var labCodeableCoding = labCodeableConcept.addCoding();
+    labCodeableCoding.setSystem(OBSERVATION_CATEGORY_CODE_SYSTEM);
+    labCodeableCoding.setCode("laboratory");
+    observation.setCategory(List.of(labCodeableConcept));
+
+    var observationCodeableConcept = new CodeableConcept();
+    var observationCoding = observationCodeableConcept.addCoding();
+    observationCoding.setSystem(LOINC_CODE_SYSTEM).setCode(props.getTestPerformedCode());
+    observation.setCode(observationCodeableConcept);
+
+    addSNOMEDValue(props.getResultValue(), observation, "");
+
+    String observationId = uuidGenerator.randomUUID().toString();
+    observation.setId(observationId);
+
+    return observation;
+  }
+
+  public Patient convertToPatient(ConditionAgnosticConvertToPatientProps props) {
+    var patient = new Patient();
+    patient.setId(props.getId());
+    parsePotentiallyAbsentName(
+        props.getFirstName(), props.getLastName(), props.getNameAbsentReason(), patient);
+    patient.setGender(convertToAdministrativeGender(props.getGender()));
+    return patient;
+  }
+
+  private Patient parsePotentiallyAbsentName(
+      String first, String last, String absentReason, Patient patient) {
+    if (StringUtils.isNotBlank(first) && StringUtils.isNotBlank(last)) {
+      var humanName = new HumanName();
+      humanName.addGiven(first);
+      humanName.setFamily(last);
+      patient.addName(humanName);
+
+    } else {
+      patient
+          .addName()
+          .addExtension(
+              new Extension()
+                  .setUrl(DATA_ABSENT_REASON_EXTENSION_URL)
+                  .setValue(new CodeableConcept().addCoding().setCode(absentReason)));
+    }
+    return patient;
+  }
+
+  public Bundle createFhirBundle(ConditionAgnosticCreateFhirBundleProps props) {
+
+    var patientFullUrl = ResourceType.Patient + "/" + props.getPatient().getId();
+    props.getDiagnosticReport().setSubject(new Reference(patientFullUrl));
+
+    var diagnosticReportFullUrl =
+        ResourceType.DiagnosticReport + "/" + props.getDiagnosticReport().getId();
+
+    UUID messageHeaderId = uuidGenerator.randomUUID();
+    var messageHeader =
+        createMessageHeader(
+            "",
```

Comment:
ditto to the comment [here](https://github.com/CDCgov/prime-simplereport/pull/6638/files#r1344629765)